### ISSUE TITLE
Tag Supports with hash and indexed search options

### ISSUE THREAD

**Issue by DrVinod** [typora/typora-issues#1393]
![](https://blogassets.evernote.com/wp-content/uploads/2014/12/Screen-Shot-2014-12-10-at-3.50.17-PM1.png)
https://zapier.com/blog/how-to-use-tags-and-labels/

**Comment by vassudanagunta** [comment]
i would love something like this. 

In the meantime, checkout http://1writerapp.com. It **indexes #hashtags**. I've spoken with the 1Writer dev, who's very responsive, about 1Writer also indexing tags in YAML front matter. He is totally down with the idea and it's on his todo list.  I tag my writings usually in front matter, but I also like to embed hashtags when I want to tag something specific deep in the content. So it would be great if they both got indexed. Fingers crossed.

Since 1Writer syncs via both iCloud and Dropbox,  it can be your mobile compliment to Typora.

**Comment by vassudanagunta** [comment]
@abnerlee This is a dupe of #1276.  But I'd say keep this one and close the other, since this has a better description, and the other references two issue, once of which is isn't an issue (Typora already supports links between source files).

@vndsrswt Do you mind if I recommend some improvements for your request? 

1. Change the title to suggesting a feature and leave implementation specifics to a comment. i.e. just "**Index tags and hashtags**"

2. In the description, start with a brief explanation, such as:

    > It would be very useful if Typora indexed YAML tags, and possibly also hashtags. Many people use Typora to manage lots of notes, or to edit content for static site generators. Many people use tags to organize large collections of pages. Static site generators  such as Jekyll and Hugo index tags, as do note taking apps such as nvALT. [See this this blog post on how to use tags and labels](url).

3. Move your idea that it go into the sidebar to the bottom of the description, or even as a reply comment. There are other possible UI solutions, but the most important thing is having a tag index, not exactly where it is made available.  

Keep the great graphic!

**Comment by borisrorsvort** [comment]
@vassudanagunta Have you still have this one your radar? 
Also, can we support you financially by buying the app or via patreon alike platforms?

**Comment by vassudanagunta** [comment]
@borisrorsvort Do you mean [1Writer](http://1writerapp.com)? I'm not the dev. But I still use the app and it is still great. But is still only indexes hashtags, not YAML tags. The dev might move it up in his priority list if more people who ask him about it (contact info on the website), especially if you buy his $3 app or put $ in his tip jar (since he doesn't charge for upgrades).

**Comment by borisrorsvort** [comment]
No I mean does typora actually plan to display a list of tags (for example in the sidebar) from YAML frontmatters?

**Comment by vassudanagunta** [comment]
I'm not the Typora dev! That would be @abnerlee.

**Comment by borisrorsvort** [comment]
@vassudanagunta Sorry about that! Thought you know because you’re so active 😛

**Comment by vassudanagunta** [comment]
@borisrorsvort I'm a big user of this currently free app, so I try to help with quality (I hope!) constructive feedback. More selfishly, I want to influence the app's evolution in a good direction. People want so many things that would ruin Typora.

**Comment by borisrorsvort** [comment]
@vassudanagunta Thanks for that :) Btw @abnerlee I would gladly pay for this app!

**Comment by borisrorsvort** [comment]
@abnerlee Hi, is this on your list? Is there a (even vague) roadmap for Typora?

**Comment by justnealpatel** [comment]
Any update on this? @abnerlee

**Comment by redoak** [comment]
The note-taking app [Notable](https://github.com/notable/notable) has a good implementation of tags. From the website:

> **Indefinitely nestable tags**: Pretty much all other note-taking apps differentiate between notebooks, tags and templates. IMHO this unnecessarily complicates things. In Notable you can have root tags (`foo`), indefinitely nestable tags (`foo/bar`, `foo/.../qux`) and it still supports notebooks and templates, they are just special tags with a different icon (`Notebooks/foo`, `Templates/foo/bar`).

In the sidebar all user defined root tags are listed as well as the special tags `Notebooks` and `Templates` (if they are used).  It also contains the special filters `All Notes`, `Tags` (all notes with any tag except `Notebooks` and `Templates`) and `Untagged` (all notes without any tags). The `Untagged` filter is quite handy when locating notes you have forgot to tag.

**Comment by vassudanagunta** [comment]
@abnerlee, can you change the title so that it is more generic and easier for people to find (@DrVinod doesn't seem to be active on GitHub anymore). 

See [my comment above](https://github.com/typora/typora-issues/issues/1393#issuecomment-380540890) for details.

**Comment by DutchPete** [comment]
Hashtags should definitely be part of the indexing feature, and in case for **Linux**, which is what I am on.

**Comment by gabexrivera** [comment]
I am also in favor of tags.  I think having both folder structure and tags makes the documents easy to find in Typora and on the filesystem.

**Comment by KorraHarraway** [comment]
Thumbs up here too for tag support!
In fact, tag support and advanced search (including words and/or tags) across different files into the parent folder would make IMO the best overall markdown note-taking app for linux.

**Comment by Donboscow** [comment]
It will be great to have tag options in typora

**Comment by raphaelhuleux** [comment]
+1

**Comment by yangxingwu** [comment]
any update?

**Comment by DrVinod** [comment]
> any update?

Not yet

**Comment by yangxingwu** [comment]
> > any update?
> 
> Not yet

@DrVinod do you have any plan or roadmap?

**Comment by yangxingwu** [comment]
any update?

**Comment by cong-han** [comment]
@DrVinod It will be great that Typora has the tags system

**Comment by vassudanagunta** [comment]
Why did this get closed? Whether or not @DrVinod wants it, so many others do.

**Comment by DrVinod** [comment]
> Why did this get closed? Whether or not @DrVinod wants it, so many others do.

Reasons are...
1. i start this app as a free for writing notes but now it is paid slowly im shifting to vs code with markdown features.
2. not getting update from long time about this feature, it is about to 4yrs...

**Comment by vassudanagunta** [comment]
Yes, but this issue is valued by other people, as you see in the comments. I don't think you should close it just because you don't care anymore. It doesn't change whether or not the issue is a real one. Your last point confirms this.

**Comment by hipages-guoxu** [comment]
Any change we can get this feature?

**Comment by oktayacikalin** [comment]
Just started using Typora some days ago and now wondering why this has not been implemented yet. Please do!

**Comment by CX330Blake** [comment]
Any update?
